### ISSUE TITLE
Documentation ambiguity about wsrep_sync_wait

### ISSUE THREAD

**Issue by GeoffMontee** [mariadb-corporation/mysql-wsrep#339]
The [documentation](http://galeracluster.com/documentation-webpages/mysqlwsrepoptions.html?highlight=wsrep_sync_wait#wsrep-sync-wait) says the following about wsrep_sync_wait:

> For example, say that you have a web application. At one point in its run, you need it to perform critical read. That is, you want the application to access the database server and run a SELECT query that must return the most up to date information possible.
> 
> SET SESSION wsrep_sync_wait=1;
> SELECT * FROM example WHERE field = "value";
> SET SESSION wsrep_sync_wait=0
> 
> In the example, the application first runs a SET command to enable wsrep_sync_wait for READ statements, then it makes a SELECT query. Rather than running the query, the node initiates a causality check, **blocking incoming queries while it catches up with the cluster**. When the node finishes applying the new transaction, it executes the SELECT query and returns the results to the application. The application, having finished the critical read, disables wsrep_sync_wait, returning the node to normal operation.

When the documentation says that galera **blocks incoming queries while it catches up with the cluster** while a session initiates a causality check, I believe this means that new queries are blocked in only the session that is performing the causality check. However, the statement is somewhat ambiguous, so some people might think that this means that **all** queries will be blocked on the node during this time. This section should probably be clarified, so that it is less ambiguous.

**Comment by mgleis** [comment]
+1 I can confirm that

**Comment by Mo33n** [comment]
+1 
Its confusing , Please update the documentation.

**Comment by marinovgit** [comment]
Which queries are blocked, only from the session that is performing the causality check or all?

**Comment by n8falke** [comment]
With `wsrep_sync_wait=1` we had a three minute break (got timeouts and waits) in the whole cluster while `LOAD DATA LOCAL` synced.

Might be it blocks all queries not only the session?

**Comment by MartinKirchner** [comment]
+1

**Comment by Chratho** [comment]
Just caught a glimpse into the source code. Seems like this feature was implemented by @sciascid or @sjaakola ? I know this was back in 2014 or 2015 but maybe you can remember which interpretation of the documentation is right?

**Comment by sciascid** [comment]
I agree, the documentation is ambiguous. I think other queries do not block.
For example, say you have clients 1 and 2, and initially `wsrep_sync_wait=0`

Client 1 does:
```
SET SESSION wsrep_sync_wait=1;
SELECT * FROM t1;  # this performs a causality check
``` 
While the causality check of client 1 is executing, client 2 does:
```
SELECT * FROM t1; # this may finish before the causality check of client 1
```

I'm pretty sure that this is how it works, but just to be sure I will verify and come back.

**Comment by sciascid** [comment]
I did verify and can confirm what I wrote in my previous comment.
I opened a ticket in our documentation repository https://github.com/codership/documentation/issues/559

**Comment by sciascid** [comment]
Fixed as part of https://github.com/codership/documentation/issues/559
